### ISSUE TITLE
Red Card

### ISSUE THREAD

**Issue by BluRosie** [BluRosie/hg-engine#259]
red card needs to handle flip turn edge case

![Image](https://github.com/BluRosie/hg-engine/assets/16446370/e85d95f7-fa84-4bb4-ab37-55b09d9e7758)

**Comment by Aeliko** [comment]
Also check edge cases with Eject Button

**Comment by BluRosie** [comment]
red card allegedly does not get consumed on first activation

**Comment by BluRosie** [comment]
yeah https://discord.com/channels/446824489045721090/1201213967389966378/1280331726802911334

**Comment by Aeliko** [comment]
Unhandled wild battle case
If held by a [wild Pokémon](https://bulbapedia.bulbagarden.net/wiki/Wild_Pok%C3%A9mon) in a Single Battle, it forces a player's Pokémon that attacks it to flee when activated. In a wild battle that was initiated as a [Horde Encounter](https://bulbapedia.bulbagarden.net/wiki/Horde_Encounter) or [Double Battle](https://bulbapedia.bulbagarden.net/wiki/Double_Battle), or an [SOS Battle](https://bulbapedia.bulbagarden.net/wiki/SOS_Battle) in which there are currently two wild Pokémon, the Red Card will cause the player's attacking Pokémon to switch rather than flee. If multiple Pokémon, each holding a Red Card, are hit by a single attack, only the fastest holder's held Red Card will activate.

Cannot verify Pickpocket + Red Card interaction because of item consumption bug
A Pokémon with [Pickpocket](https://bulbapedia.bulbagarden.net/wiki/Pickpocket_(Ability)) holding a Red Card can steal an attacker's item with Pickpocket even after that attacker is switched out by the Pokémon's held Red Card; if a Pokémon steals a Red Card due to Pickpocket, that Red Card cannot activate for that same attack. The Red Card switches out the attacker before its own held [Life Orb](https://bulbapedia.bulbagarden.net/wiki/Life_Orb) or [Shell Bell](https://bulbapedia.bulbagarden.net/wiki/Shell_Bell) has a chance to activate. If a Pokémon holding a Red Card is hit by a combination move ([Grass Pledge](https://bulbapedia.bulbagarden.net/wiki/Grass_Pledge_(move)), [Fire Pledge](https://bulbapedia.bulbagarden.net/wiki/Fire_Pledge_(move)), or [Water Pledge](https://bulbapedia.bulbagarden.net/wiki/Water_Pledge_(move))) without fainting, and the attacker is switched out, the combination effect will not be created. If a Pokémon holding a Red Card is hit by [Dragon Tail](https://bulbapedia.bulbagarden.net/wiki/Dragon_Tail_(move)) or [Circle Throw](https://bulbapedia.bulbagarden.net/wiki/Circle_Throw_(move)) without fainting, it is forced to switch out or flee without giving its Red Card the opportunity to activate. If a Pokémon holding a Red Card is hit by [U-turn](https://bulbapedia.bulbagarden.net/wiki/U-turn_(move)), [Volt Switch](https://bulbapedia.bulbagarden.net/wiki/Volt_Switch_(move)) or [Flip Turn](https://bulbapedia.bulbagarden.net/wiki/Flip_Turn_(move)) without fainting, the Red Card activates and forces the attacker to switch to a random Pokémon, instead of a Pokémon of their Trainer's choice. [Emergency Exit](https://bulbapedia.bulbagarden.net/wiki/Emergency_Exit_(Ability)) and [Wimp Out](https://bulbapedia.bulbagarden.net/wiki/Wimp_Out_(Ability)) will not activate if a Red Card is used up on the same move, while U-turn, Volt Switch and Flip Turn will not cause the user to switch if they are holding a Red Card.

**Comment by Pixelstyx** [comment]
Kuma's PR: https://github.com/BluRosie/hg-engine/pull/741
